src/nupic/algorithms/SpatialPooler.hpp

```diff
@@ -66,7 +67,7 @@ namespace nupic {
        *     }
        *
        */
-      class SpatialPooler {
+      class SpatialPooler : public nupic::Serializable<SpatialPoolerProto> {
```

Comment:
we are already inside the `nupic` namespace so you shouldn't need to qualify `Serializable`